plugins/tiddlywiki/prosemirror/setup/menu.js

```diff
@@ -120,32 +122,44 @@ function wrapListItem(nodeType, options) {
 function buildMenuItems(schema) {
 	var r = {};
 	var mark;
-	if (mark = schema.marks.strong)
+	mark = schema.marks.strong;
+	if(mark)
 		r.toggleStrong = markItem(mark, {title: "Toggle strong style", icon: icons.strong});
-	if (mark = schema.marks.em)
+	mark = schema.marks.em;
+	if(mark)
 		r.toggleEm = markItem(mark, {title: "Toggle emphasis", icon: icons.em});
-	if (mark = schema.marks.code)
+	mark = schema.marks.code;
+	if(mark)
 		r.toggleCode = markItem(mark, {title: "Toggle code font", icon: icons.code});
-	if (mark = schema.marks.link)
+	mark = schema.marks.link;
+	if(mark)
 		r.toggleLink = linkItem(mark);
 
 	var node;
-	if (node = schema.nodes.image)
+	node = schema.nodes.image;
+	if(node)
 		r.insertImage = insertImageItem(node);
-	if (node = schema.nodes.bullet_list)
+	node = schema.nodes.bullet_list;
+	if(node)
 		r.wrapBulletList = wrapListItem(node, {title: "Wrap in bullet list", icon: icons.bulletList});
-	if (node = schema.nodes.ordered_list)
+	node = schema.nodes.ordered_list;
+	if(node)
 		r.wrapOrderedList = wrapListItem(node, {title: "Wrap in ordered list", icon: icons.orderedList});
-	if (node = schema.nodes.blockquote)
+	node = schema.nodes.blockquote;
+	if(node)
 		r.wrapBlockQuote = wrapItem(node, {title: "Wrap in block quote", icon: icons.blockquote});
-	if (node = schema.nodes.paragraph)
+	node = schema.nodes.paragraph;
+	if(node)
 		r.makeParagraph = blockTypeItem(node, {title: "Change to paragraph", label: "Plain"});
-	if (node = schema.nodes.code_block)
+	node = schema.nodes.code_block;
+	if(node)
 		r.makeCodeBlock = blockTypeItem(node, {title: "Change to code block", label: "Code"});
-	if (node = schema.nodes.heading)
-		for (var i = 1; i <= 10; i++)
+	node = schema.nodes.heading;
+	if(node)
```

Comment:
Please use curly braces. This construction is extra brittle.

Reply:
I copy this from prosemirror example, the original code is even worser 

https://github.com/ProseMirror/prosemirror-example-setup/blob/8c11be6850604081dceda8f36e08d2426875e19a/src/inputrules.ts#L44-L46

But adding curly brace will make it much longer, do you mean add it in same line?

Reply:
The main focus should be maintainability and human understandability. If the code is a 100 lines more does not really matter. It will only be 2 lines more per if() but maintainability it will pay off in the long run.